x-pack/plugins/dataset_quality/server/routes/data_streams/routes.ts

```diff
@@ -28,12 +28,16 @@ const statsRoute = createDatasetQualityServerRoute({
     tags: [],
   },
   async handler(resources): Promise<DataStreamsStatResponse> {
-    const { context, params } = resources;
+    const { context, params, plugins } = resources;
     const coreContext = await context.core;
 
     // Query datastreams as the current user as the Kibana internal user may not have all the required permissions
     const esClient = coreContext.elasticsearch.client.asCurrentUser;
 
+    const fleetPluginStart = await plugins.fleet.start();
+    const packageClient = fleetPluginStart.packageService.asInternalUser;
```

Comment:
What do you mean? could you give me an example? are you talking about a service?

Reply:
We cannot expose the client already since it will depend on the type of permissions we want to use to query the information, for this particular endpoint `asInternalUser` is just enough but for some other case we might want to query as `currentUser`, but we could expose sort of a service from plugin setup/start and use it in our routes. I see a trade off though: every time we include a new dependency we would need to expose a service whereas using the plugins in this way will give us the plugin `for free`, just awaiting for it to be started, this awaiting time will most likely be higher at the very first start of the plugin.

Reply:
As agreed we will go with this approach and revisit later if a service will suit better our case.